### PR TITLE
bpo-41384: fix .keys()[0] in tkinter

### PR DESCRIPTION
**Pull request by Akuli**
not sure if this needs bugs.python.org number or not

https://bugs.python.org/issue41384

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@Akuli

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by csabella** [comment]
Yes, please open an issue on the bug tracker.  Thank you!

**Comment by Akuli** [comment]
do I really need to write my home address to an agreement in order to create a small bug fix pull request?

**Comment by tirkarthi** [comment]
Code changes required CLA. Though the fix is trivial I guess this is a bugfix and would require a CLA and NEWS entry.

**Comment by Akuli** [comment]
What should I do if I really don't want to sign the CLA? I don't think this bug actually bothers anyone because getting the wrong kind of error isn't too bad, not many people are using tkinter and not many tkinter users are using OptionMenu.

File: Lib/tkinter/__init__.py [hunk 1]
```diff
@@ -3965,7 +3965,7 @@ def __init__(self, master, variable, value, *values, **kwargs):
         if 'command' in kwargs:
             del kwargs['command']
         if kwargs:
-            raise TclError('unknown option -'+kwargs.keys()[0])
+            raise TclError('unknown option -'+next(iter(kwargs.keys())))
```
Comment (serhiy-storchaka) [inline Lib/tkinter/__init__.py]: `.keys()` is not needed.
```suggestion
            raise TclError('unknown option -'+next(iter(kwargs)))
```

Comment (Akuli) [reply Lib/tkinter/__init__.py]: I preferred being explicit here, but I'm happy with it either way.

**Review by serhiy-storchaka — APPROVED** [review]
:+1:

File: Lib/tkinter/test/test_tkinter/test_widgets.py [hunk 2]
```diff
@@ -307,6 +307,10 @@ class OptionMenuTest(MenubuttonTest, unittest.TestCase):
     def create(self, default='b', values=('a', 'b', 'c'), **kwargs):
         return tkinter.OptionMenu(self.root, None, default, *values, **kwargs)
 
+    def test_bad_kwarg(self):
+        with self.assertRaisesRegex(TclError, r"^unknown option -image$"):
+            tkinter.OptionMenu(self.root, None, 'b', 'a', 'b', 'c', image='::tk::icons::question')
```
Comment (terryjreedy) [inline Lib/tkinter/test/test_tkinter/test_widgets.py]: Remove irrelevant and distracting detail from call.
```suggestion
            tkinter.OptionMenu(self.root, None, 'b', image='')
```

**Comment by terryjreedy** [comment]
Why would be your objection to signing the CLA?

**Comment by Akuli** [comment]
I don't want to associate my github user with my home address. Why do I need to tell you (or Adobe or whoever sees the signed CLA) where I live in order to contribute a small bugfix?

**Comment by serhiy-storchaka** [comment]
I think this change does not require signing the CLA. It is trivial enough.

**Comment by terryjreedy** [comment]
Akuli, PSF does not have the resources that IBM and Google did to fight the Unix and Java API lawsuits.  So we try to exercise 'due diligence'.

Having done so, I think*, I will merge this PR.  2to3 should have wrapped `kwargs.keys()` with `list()`, and that would have worked, but using `iter` is the documented way to access an arbitrary key.  The code fix is too trivial for copyright.  The added test is marginal, but it follows a template and you included my suggestion.

* If you had said, "Because I wrote this at work" or "I disagree with the CLA", I would have replied "Thank you, but...", closed this, and written a new PR with the alternate fix.

**Comment by Akuli** [comment]
I get `@` mentioned every once in a while because someone screws up a `git rebase` and the bot reminds about my missing CLA. For now I'll try to silence all notifications coming from this repo. Please email me if there's something important.

**Pull request merged**